script.js

```diff
@@ -0,0 +1,44 @@
+let saladMenuFileData = [], prices = [];

```

Comment:
Kiedyś deklarowanie zmiennych po przecinku było pewnego rodzaju standardem. Teraz się od tego odchodzi i słusznie bo dużo lepiej się czyta kod kiedy jedna linijka odpowiada za jedną czynność. Druga sprawa jest taka, że let jest "mutowalny". Popatrz sobie wzorzec immutability to jest część paradygmatu programowania funkcyjnego. Jeżeli nie musisz przeciążać zmiennej inną wartością to używaj const. Druga sprawa jest taka, że metody arrayki (jak push) i obiektu (jak assign) są mutowalne i pomimo używania const na arrayce lub obiekcie możesz zmienić ich zawartość, ale nie możesz jawnie przeciążyć zmiennej nową wartością. Przykład:
1. Weźmy sobie arrayke:
const salads = ["z kurczaka", "gyros"];

Dodajmy jeden element;
salads.push("z ananasem");

Teraz arrayka ma 3 elementy. Interpreter nie krzyczy i jest to poprawne dodanie elementu do arrayki. Mimo użycia const zmieniliśmy ilość elementów w arrayce. Teraz dodajmy przeciążając wartość zmiennej.

salads = [...salads, "z ananasem"];

I teraz dostaniemy error informujący nas o tym, że chcemy zmienić wartość. Tak nie możemy zrobić. Swoją drogą syntax "..." to spread/rest operator. Poczytaj sobie o nim jak go nie znasz, jest bardzo przydatny. Jeżeli użyjemy let zamiast const taka operacja będzie poprawna. Natomiast jeżeli chcemy czerpać z programowania funkcyjnego to mutability sugeruje nam żeby unikać mutowania stanu zmiennej. Dlatego zaleca się stworzenie nowej zmiennej jak mutujemy inną:
const updatedSalads = [...salads, "z ananasem"];

Zazwyczaj nie dodaje się z palca elementów arrayki tylko elementy przychodzą z api z backendu.